compiler/src/iree/compiler/Codegen/Common/TileSizeSelection.h

```diff
@@ -15,6 +15,75 @@ namespace mlir::iree_compiler {
 using SizesAndScalableFlags =
     std::pair<SmallVector<int64_t>, SmallVector<bool>>;
 
+using SizeAndScalableFlag = std::tuple<int64_t &, bool &>;
+
+/// A tuple that encapsulates two quantities describing tile sizes:
+///   * regular tile sizes (integers) - that's always required
+///   * scalable tile flags (bool - only used/required for scalable
+///     vectorisation.
+/// Use this wrapper to make sure that both quantities are upated when
+/// manipulating tile sizes.
+struct SizesAndScalableFlagsTuple {
+  SmallVector<int64_t> sizes;
+  SmallVector<bool> flags;
+
+  // Represents a pair of references to a size and a scalable flag at the given
+  // index. Due to various implementation details of vector of bools, it's much
+  // easier to store a reference to a whole container and an index. While this
+  // increases the size of this wrapper, it also simplifies the implementation.
+  struct ReferencePair {
+    SmallVector<int64_t> &sizesVec;
+    SmallVector<bool> &flagsVec;
+    // Index of this pair within the vectors
+    size_t index;
+
+    explicit ReferencePair(const ReferencePair &a) = default;
+    ReferencePair(SmallVector<int64_t> &sizesVecRef,
+                  SmallVector<bool> &boolVectorRef, size_t indexRef)
+        : sizesVec(sizesVecRef), flagsVec(boolVectorRef), index(indexRef) {}
+
+    // Update this pair based on the input integer + bool
+    ReferencePair &operator=(const std::pair<int64_t, bool> &values) {
+      sizesVec[index] = values.first;
+      flagsVec[index] = values.second;
+      return *this;
+    }
+
+    // Update this pair based on the input integer. Assume that the scalable
+    // size is false. This is safe to use in cases where no scalable
+    // vectorisation/tiling is used/supported.
+    ReferencePair &operator=(int64_t size) {
+      sizesVec[index] = size;
+      flagsVec[index] = false;
+      return *this;
+    }
+
+    // Update this pair based on the input ReferencePair
+    ReferencePair &operator=(const ReferencePair &pair) {
+      sizesVec[index] = pair.sizesVec[index];
+      flagsVec[index] = pair.flagsVec[index];
+      return *this;
+    }
+  };
+
+  SizesAndScalableFlagsTuple(SmallVector<int64_t> s, SmallVector<bool> f)
+      : sizes(s), flags(f) {}
+
+  // Initialise to {0, false} for all sizes
+  SizesAndScalableFlagsTuple(size_t numElements)
+      : sizes(SmallVector<int64_t>(numElements, 0)),
+        flags(SmallVector<bool>(numElements, false)) {}
```

Comment:
I think we don't need `SmallVector<...>` to construct it. Does the below snippet work?

```suggestion
  SizesAndScalableFlagsTuple(size_t numElements)
      : sizes(numElements, 0),
        flags(numElements, false) {}
```